### ISSUE TITLE
加载预训练模型finetune自定义数据集，infer会出现额外的类别

### ISSUE THREAD

**Issue by bbbxixixixi** [PaddlePaddle/PaddleClas#998]
配置文件中只有3类，输出却会有其他类别的信息
# global configs
Global:
  checkpoints: null
  pretrained_model: /home/aistudio/ResNeXt50_vd_64x4d_pretrained
  output_dir: /home/aistudio/output/
  device: gpu
  class_num: 3
  save_interval: 1
  eval_during_train: True
  eval_interval: 1
  epochs: 100
  print_batch_step: 10
  use_visualdl: True
  # used for static mode and model export
  image_shape: [3, 448, 448]
  save_inference_dir: /home/aistudio/inference

# model architecture
Arch:
  name: ResNeXt50_vd_64x4d
 
# loss function config for traing/eval process
Loss:
  Train:
    - CELoss:
        weight: 1.0
        epsilon: 0.1
  Eval:
    - CELoss:
        weight: 1.0

Optimizer:
  name: Momentum
  momentum: 0.9
  lr:
    name: Cosine
    learning_rate: 0.01
  regularizer:
    name: 'L2'
    coeff: 0.0001

# data loader for train and eval
DataLoader:
  Train:
    dataset:
      name: ImageNetDataset
      image_root: /home/aistudio/data/train
      cls_label_path: /home/aistudio/data/train_splited_txt/f5_1/train.txt
      transform_ops:
        - DecodeImage:
            to_rgb: True
            channel_first: False
        - RandCropImage:
            size: 448
        - RandFlipImage:
            flip_code: 1
        - NormalizeImage:
            scale: 1.0/255.0
            mean: [0.485, 0.456, 0.406]
            std: [0.229, 0.224, 0.225]
            order: ''
      batch_transform_ops:
        - MixupOperator:
            alpha: 0.2

    sampler:
      name: DistributedBatchSampler
      batch_size: 32
      drop_last: True
      shuffle: True
    loader:
      num_workers: 1
      use_shared_memory: False

  Eval:
    dataset: 
      name: ImageNetDataset
      image_root: /home/aistudio/data/train
      cls_label_path: /home/aistudio/data/train_splited_txt/f5_1/eval.txt
      transform_ops:
        - DecodeImage:
            to_rgb: True
            channel_first: False
        - ResizeImage:
            resize_short: 448
        - CropImage:
            size: 448
        - NormalizeImage:
            scale: 1.0/255.0
            mean: [0.485, 0.456, 0.406]
            std: [0.229, 0.224, 0.225]
            order: ''
    sampler:
      name: DistributedBatchSampler
      batch_size: 32
      drop_last: False
      shuffle: False
    loader:
      num_workers: 1
      use_shared_memory: False

Infer:
  infer_imgs: /home/aistudio/data/train/1/76dff756-b3d8-4d84-b5a9-5d748bc1545f.jpg
  batch_size: 10
  transforms:
    - DecodeImage:
        to_rgb: True
        channel_first: False
    - ResizeImage:
        resize_short: 448
    - CropImage:
        size: 448
    - NormalizeImage:
        scale: 1.0/255.0
        mean: [0.485, 0.456, 0.406]
        std: [0.229, 0.224, 0.225]
        order: ''
    - ToCHWImage:
  PostProcess:
    name: Topk
    topk: 5
    class_id_map_file: /home/aistudio/data/train_splited_txt/f5_1/train.txt

Metric:
  Train:
  Eval:
    - TopkAcc:
        topk: [1, 5]

输出class_ids有别的东西
[{'class_ids': [3, 2, 1, 980, 974], 'scores': [0.36787, 0.3455, 0.18132, 0.00012, 0.00012], 'file_name': 'aaaa/011711d3-0f63-4346-8fb4-64351db59950.jpg', 'label_names': []}, {'class_ids': [3, 2, 1, 980, 974], 'scores': [0.35961, 0.31867, 0.21217, 0.00013, 0.00013], 'file_name': 'aaaa/6a35ae63-0a23-46a3-80c4-98be8e5066ff.jpg', 'label_names': []}, {'class_ids': [2, 3, 1, 972, 980], 'scores': [0.38646, 0.3687, 0.14165, 0.00012, 0.00012], 'file_name': 'aaaa/75f22f34-45ff-4aba-9e06-831195d9d374.jpg', 'label_names': []}, {'class_ids': [3, 1, 2, 304, 324], 'scores': [0.51291, 0.31641, 0.14388, 0.0002, 0.00019], 'file_name': 'aaaa/844be6c6-a90c-42d8-8463-e9eca90cc9b6.jpg', 'label_names': []}, {'class_ids': [3, 1, 2, 29, 103], 'scores': [0.3423, 0.28713, 0.26282, 0.00014, 0.00014], 'file_name': 'aaaa/abada0bd-8eec-429f-8038-bab866c47888.jpg', 'label_names': []}, {'class_ids': [3, 1, 2, 304, 324], 'scores': [0.51291, 0.31641, 0.14388, 0.0002, 0.00019], 'file_name': 'aaaa/efd94775-e86f-4c6e-b589-3af6733b8ade.jpg', 'label_names': []}]

**Comment by cuicheng01** [comment]
您好，之前这里有一个小问题，导致Global.class_num没有生效，最后的FC默认是1000，现在最新的代码设置Arch.class_num重新训练或者finetune即可。

**Comment by bbbxixixixi** [comment]
> 您好，之前这里有一个小问题，导致Global.class_num没有生效，最后的FC默认是1000，现在最新的代码设置Arch.class_num重新训练或者finetune即可。

我用的是release2.2，这个版本设置Arch.class_num就行了对吧？

**Comment by cuicheng01** [comment]
> > 您好，之前这里有一个小问题，导致Global.class_num没有生效，最后的FC默认是1000，现在最新的代码设置Arch.class_num重新训练或者finetune即可。
> 
> 我用的是release2.2，这个版本设置Arch.class_num就行了对吧？

之前的有问题，现在修正了，拉取最新的2.2是没问题的
